### DISCUSSION TITLE
HammerDB not responding with MS SQL Windows Host Cluster

### DISCUSSION THREAD

**Discussion by kayvwar** [TPC-Council/HammerDB#628] [Q&A] [answered]
Hi team,

Is HammerDB supported with MS SQL in a Windows Host Cluster config.
I have a 2 host Windows Host Cluster configured with Cluster Shared Volumes.
I have installed SQL in a clustered manner.
A test database is created and initialized by HammerDB utility.
I am able to run load as well.
The challenge is when I initiate the cluster failover. For example I restart the host which owns the cluster disks, Windows failover takes place. I can see the disk ownership move to the other host. 
The database is up and running as well.
However the load that was underway using hammerdb freezes up. I  see HammerDB not responding.
Only way out is to kill the application.
The SQL database is access using its cluster IP address and not the hosts local IP address.
It is the same behaviour irrespective of quantum of IO.  Could you please let me know if HammerDB is expected to work with SQL in a Windows Failover CLuster setup (with failover event).

**Comment by sm-shaw** [comment, marked as answer]
This is a similar question to what is described here: https://github.com/TPC-Council/HammerDB/issues/573
So HammerDB has not been designed with the intention of users terminating the database mid-test, however there are a number of options that give flexibility in testing clusters and it looks like other people have done something similar. 
If the database fails mid-test however then the connections should be terminated and not hang so this may be something in your cluster configuration. 
https://yarondar.wordpress.com/2021/09/19/using-powermax-snapvx-with-sql-server-2019-and-windows-server-failover-cluster-part-2-hammerdb/
In particular, the iterations value in the virtual user options will cause the virtual user to log back in and re-run the same workload if the connection is terminated. Also, the connect pool option allows connecting to different nodes in a cluster at the same time. 
Another consideration is using the asynchronous connections with keying and thinking time so the throughput is not as high.
